src/pages/new-chat/new-chat.ts

```diff
@@ -9,22 +12,30 @@ import { User } from 'api/models/whatsapp-models';
   templateUrl: 'new-chat.html'
 })
 export class NewChatComponent implements OnInit {
-  users;
+  searchPattern = '';
   senderId: string;
+  users: Observable<User[]>;
 
   constructor(
-    public navCtrl: NavController,
-    public viewCtrl: ViewController,
-    public alertCtrl: AlertController
+    private navCtrl: NavController,
+    private viewCtrl: ViewController,
+    private alertCtrl: AlertController
   ) {
     this.senderId = Meteor.userId();
   }
 
   ngOnInit() {
-    MeteorObservable.subscribe('users').subscribe(() => {
-      MeteorObservable.autorun().subscribe(() => {
-        this.users = this.findUsers().zone();
-      });
+    this.findContacts().then((contacts) => {
+      // Pluck all contacts' phone numbers into a flat array
+      contacts = contacts && _.chain(contacts)
```

Comment:
In case we're using the browser, and not a mobile device, the `contacts` variable should be undefined. I could also use a default assignment (`contacts = []`), but it wouldn't prevent the chaining logic from being invoked. I just wanted to save some processing time.